pyproject.toml

```diff
@@ -47,6 +51,9 @@ testpaths = [
 
 [tool.poe.tasks]
 lint = "ruff check ."
+typecheck = "mypy ."
+format = "black . --check"
+format-fix = "black ."
```

Comment:
I would suggest to have a single task to run the all (mypy, black and linting). It makes it easier to run them locally with a single command, which is a bit convenient and don't forget to run the others

Reply:
added a slinge command to run all tasks
kept it separated in the github action, but open for discussion